stablehlo/dialect/TypeInference.cpp

```diff
@@ -3289,6 +3289,32 @@ LogicalResult verifyBroadcastInDimOp(std::optional<Location> location,
   return success();
 }
 
+
+LogicalResult verifyCollectiveBroadcastOp(
```

Comment:
Proposed change:
```
LogicalResult verifyCollectiveBroadcastOp(std::optional<Location> location,
                                          DenseIntElementsAttr replicaGroups) {
  // collective_permute_i2
  auto replicaGroupType = replicaGroups.getType().cast<RankedTensorType>();
  if (replicaGroupType.getRank() != 2)
    return emitOptionalError(location,
                             "replica groups should be a rank 2 tensor");
  
  auto replicaIds = replicaGroups.getValues<int64_t>();
  llvm::SmallSet<int64_t, 8> replicaIdsSeen;
  for (int64_t replicaId : replicaIds) {
    // collective_broadcast_c1
    if (replicaId < -1)
      return emitOptionalError(
          location, "replica_groups values must be positive but got ",
          replicaId);

    // collective_broadcast_c1
    if (!replicaIdsSeen.insert(replicaId).second)
      return emitOptionalError(location, "replica id #", replicaId,
                               " seen more than once");
  }

  return success();
}
```

Feedback/Notes:
* Following similar coding style within the file.
* Clearly marking each constraint (to be) tested.
* Here I'm checking `replicaId < -1` since `-1` is used to represent ragged tensors. E.g. `replica_groups = dense<[[0,1,2,3], [4,5,-1,-1]]> : tensor<2x4xi64>`.
* Verifying (C2) is infeasible since we do not know `num_replicas` or `num_partitions` statically.
* (C3) is covered by ODS, no further tests needed.